display/ssd1306.h

```diff
@@ -468,22 +474,80 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
 	  ShowFile(&IMG_idle, 3600000.0);
 	}
 	return;
+       case EFFECT_LOCKUP_BEGIN:
+        ShowDefault();
+        return;
+      case EFFECT_LOCKUP_END:
+        ShowDefault(true);
+        return;
+      case EFFECT_BATTERY_LEVEL:
+        // Show battery meter for ProffieOSFontImageDuration??
+        SetScreenNow(SCREEN_PLI);
+        return;
+#ifndef OLED_SYNCED_EFFECTS // a better name is welcomed
+      case EFFECT_FORCE:
+        ShowFile(&IMG_force, font_config.ProffieOSForceImageDuration);
+        return;
       case EFFECT_BLAST:
-	ShowFile(&IMG_blst, font_config.ProffieOSBlastImageDuration);
-	return;
+        ShowFile(&IMG_blst, font_config.ProffieOSBlastImageDuration);
+        return;
       case EFFECT_CLASH:
-	ShowFile(&IMG_clsh, font_config.ProffieOSClashImageDuration);
-	return;
+        ShowFile(&IMG_clsh, font_config.ProffieOSClashImageDuration);
+        return;
+      case EFFECT_RELOAD:
+        ShowFile(&IMG_reload, font_config.ProffieOSReloadImageDuration);
+        return;
+      case EFFECT_EMPTY:
+        ShowFile(&IMG_empty, font_config.ProffieOSEmptyImageDuration);
+        return;
+      case EFFECT_JAM:
+        ShowFile(&IMG_jam, font_config.ProffieOSJamImageDuration);
+        return;
+      case EFFECT_CLIP_IN:
+        ShowFile(&IMG_clipin, font_config.ProffieOSClipinImageDuration);
+        return;
+      case EFFECT_CLIP_OUT:
+        ShowFile(&IMG_clipout, font_config.ProffieOSClipoutImageDuration);
+        return;
+#endif
+      break;
+      default: break;
+    }
+  }
+ 
+  void SB_Effect2(EffectType effect, float location) override {
+    switch (effect) {
+      case EFFECT_DESTRUCT:
+        ShowFile(&IMG_destruct, SaberBase::sound_length * 1000);
+        return;
+        // STDOUT.println(SaberBase::sound_length * 1000);
+#ifdef OLED_SYNCED_EFFECTS
```

Comment:
Instead of having a define for this, a config variable might be better.
Not sure if that config variable would need to be per-effect or not though....

Or maybe we use the sound length only if the relevant duration variable is unset?

Reply:
In the previous PR, That's what I thought too. A setting of zero in the config.ini could indicate to use WavLen.
I had typed this:


> instead of another define, I tried something like this where the user could set the config.ini value to zero and it would use wavlen.
Then I thought maybe just commenting it out in config.ini would be better, but I don't know how to check if it was read from the file as a VariableOp or not.

  case EFFECT_CLASH:
    ShowFile(&IMG_clsh, (font_config.ProffieOSClashImageDuration = 0) ? SaberBase::sound_length * 1000 : **iterateVariables VariableOp-values somehow goes here** );
    return;
   
  float dur;
  case EFFECT_BLAST:
    if (!font_config.ProffieOSBlastImageDuration) {         // can't check if this entry exists this way
      dur = (SaberBase::sound_length * 1000);
    } else { 
      dur = font_config.ProffieOSBlastImageDuration;
    }
    ShowFile(&IMG_blst, dur);

Reply:
or a hybrid of those 2. 
Can we default the time to zero
`CONFIG_VARIABLE2(ProffieOSBlastImageDuration, 0.0f);`
 and then have the config.ini part describe like
 ```
# These are the durations the images will play for.
# Enter a custom value to manually set a duration,
# or leave commented out to automatically sync to the effect sound.
# ProffieOSBlastImageDuration=1000.0
```
float dur;
case EFFECT_BLAST:
if (ProffieOSBlastImageDuration = 0) {
    dur = (SaberBase::sound_length * 1000);
} else {
dur = font_config.ProffieOSBlastImageDuration;
}
ShowFile(&IMG_blst, dur);
```